docklets/Trash/TrashDockItem.vala

```diff
@@ -91,7 +91,7 @@ namespace Docky
 			if (item_count == 0U)
 				Text = _("No items in Trash");
 			else
-				Text = ngettext ("%u item in Trash", "%u items in Trash", item_count).printf (item_count);
+				Text = ngettext (_("%u item in Trash"), _("%u items in Trash"), item_count).printf (item_count);
```

Comment:
Unfortunately this is not how `ngettext` works. It does work with the current notation―without `_(`.

Reply:
> Unfortunately this is not how `ngettext` works. It does work with the current notation―without `_(`.

Please check the changed files section. I have already done this change.

Reply:
I have checked how it works and I can say that `ngettext` is the plural version of `gettext` I believe that it works fine. I  guess nothing wrong with this update.

Reply:
As @ryonakano says, this is not the correct way to resolve this. `dockets/Trash/TrashDockItem.vala` needs adding to `po/POTFILES.in`, then meson should be able to make the necessary changes to the translation templates itself.

Reply:
`ngettext` has three arguments: the first string is used for singular form, the second one is for plural form, and the third one is used to determine which string to show. In English, we use singular only when the number of the object we want to count is 1 and plural for other numbers. However, some languages have multiple plural forms, for example, Russian has 2 plural forms. So, `ngettext` prepares 3 entries (1 singular form + 2 plural forms) in the translation file, like this:

```po
#: ../docklets/Trash/TrashDockItem.vala:94
msgid "%u item in Trash"
msgid_plural "%u items in Trash"
msgstr[0] "" # For singular form
msgstr[1] "" # For the first plural form
msgstr[2] "" # For the second plural form
```

However, if we encloses the first and second argument of `ngettext`,  it does detect these strings as independent strings and prepares independent entries, like this:

```po
#: ../docklets/Trash/TrashDockItem.vala:94
msgid "%u item in Trash"
msgstr ""

#: ../docklets/Trash/TrashDockItem.vala:94
msgid "%u items in Trash"
msgstr ""
```

This causes an issue that in some languages like Russian it's no longer possible to translate these strings into multiple plural forms. So, we may not want to add extra `_(…)`.

Reply:
If I reverted it and added the singular/plural forms into PO file, would it be fixed? I'm asking in order to get a better understanding of it. The other question is would it be an acceptable fix?

Reply:
Nope, the templates for the singular/plural forms are automatically added to the PO files, then translations are usually done in weblate. Though I'm not sure if weblate is set up for this repository yet, as it's pretty new.

Please make the changes we've said to `POTFILES.in` and take a look at https://elementary.io/docs/code/getting-started#Adding-Translations to see how meson/ninja handle translations (especially steps 8 and 9)/

Reply:
@davidmhewitt, I have already checked the document that you have mentioned above but I have not able to see Dock setup in Weblate that's why I had added them. Should I wait for a Weblate setup for Dock?

Reply:
Even when weblate is set up, these strings will still not be translatable because `dockets/Trash/TrashDockItem.vala` needs adding to `po/POTFILES.in`. Then you can use ninja/meson to make changes to the PO/POT files for all languages.

Reply:
I got it. I was just trying to say, even if I have made those changes like adding that Vala file into POTFILES.ini, I should still wait for Weblate setup, shouldn't I? By the way, thanks both of you for your precious information.